Automattic/wp-calypso
# Social previews | Add Nextdoor preview

## Proposed Changes

* This PR adds nextdoor.com preview to Social Previews package

## Testing Instructions

- Goto `/posts/:site`
- Click on actions menu (3 dots) for any post
- Click on _"Preview"_
- Confirm that you DO NOT see the Nextdoor section in the sidebar
- Add the blog sticker to your site:
`add_blog_sticker( 'jetpack-social-nextdoor-connection', null, null, <YOUR_SITE_ID> );` 
- Goto `/marketing/connections/:site` and add a Nextdoor connection for the site
- Goto `/posts/:site`
- Click on actions menu (3 dots) for any post
- Click on _"Preview"_
- Confirm that you see the Nextdoor section in the sidebar
- Click on Nextdoor
- Confirm that you see the Nextdoor post preview
- Confirm that the preview matches the link previews on nextdoor.com
- Confirm the preview for different scenarios for a post
    * with and without featured image
    * with featured image and custom media
    * without featured image but with custom media
    * with video as custom media

In Calypso
<img width="1456" alt="Screenshot 2023-11-01 at 12 33 58 PM" src="https://github.com/Automattic/wp-calypso/assets/18226415/47f0f3c8-f057-467f-b2ce-dd851ffe537a">

In Jetpack


- With featured image and video as custom media
    <img width="948" alt="Screenshot 2023-11-01 at 11 15 50 AM" src="https://github.com/Automattic/wp-calypso/assets/18226415/cfafcb97-f834-4e5d-9cbe-99874698c366">
- With featured image and photo as custom media
    <img width="956" alt="Screenshot 2023-11-01 at 11 17 36 AM" src="https://github.com/Automattic/wp-calypso/assets/18226415/a62f2486-8324-46ff-aafe-8ab08c55a63b">
- With featured image and photo as custom media but "Share as a social post" disabled
    <img width="936" alt="Screenshot 2023-11-01 at 11 18 31 AM" src="https://github.com/Automattic/wp-calypso/assets/18226415/e1fc418a-b30a-423d-8b69-46e84e7884e6">
- With featured image but without custom media
    <img width="941" alt="Screenshot 2023-11-01 at 11 19 04 AM" src="https://github.com/Automattic/wp-calypso/assets/18226415/cbd0cfd4-88c1-4251-8925-c7ecf5a50bf3">
- Without featured image and without custom media
    <img width="931" alt="Screenshot 2023-11-01 at 11 51 14 AM" src="https://github.com/Automattic/wp-calypso/assets/18226415/bba1ed29-c45f-4678-be35-2fe43b245601">
- Without featured image and with video as custom media
    <img width="934" alt="Screenshot 2023-11-01 at 12 07 36 PM" src="https://github.com/Automattic/wp-calypso/assets/18226415/21004a8c-9b84-4498-878e-4ae78b4f23d0">

## Pre-merge Checklist
